contracts/initializable/BondingCurve3.sol

```diff
@@ -114,9 +115,12 @@ contract BondingCurve3 is IERC1363Spender {
             require(ShardedWallet(payable(wallet_)).transferFrom(msg.sender, address(this), supply));
         }
 
-        // setup curve
-        curve.x = totalSupply_;
-        curve.k = totalSupply_ * totalSupply_ * price / 10**18;
+        {
+            // setup curve
+            uint256 decimals_ = ShardedWallet(payable(wallet_)).decimals();
```

Comment:
sharded wallet are 18 decimals, and have to simple way of changing that.
Querying the decimals as a gas cost that we can avoid by just assuming 18 decimals palces

Reply:
This is to ensure consistency in case someone else use this contract and change decimals.
Agreed this costs a bit gas for cross-contract call